persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedActorOf.scala

```diff
@@ -172,15 +131,49 @@ object EventSourcedActorOf {
 
   }
 
-  implicit private class CriteriaOps(val criteria: SnapshotSelectionCriteria) extends AnyVal {
+  implicit final private class SnapshotStoreOps[F[_], A](val store: SnapshotStore[F, A]) extends AnyVal {
```

Comment:
I wonder why this is a method in an implicit class, rather than just a private method in `SnapshotStore`.

I am not against it, and I seen @t3hnar doing the similar thing, but I wonder what is the reasoning behind it. Does it affect backwards compatibility or something?

Reply:
My motivation is in decoupling `SnapshotStore` (API used by persistence implementations) and `Snapshotter` (internal API used by akka-effect & co) traits. I see them as independent parts from different abstraction levels thus tried not coupling them

Reply:
Does it make sense to make a simple method then?

I.e. something like this:
```
def snapshotter[F[_], A](store: SnapshotStore[F, A]): Snapshotter[F, A]
```

I might be a little bit old-fashioned in that sense, but it feels counter-intuitive to me to have an implicit class for sake of adding a single private method, which will only be used once.

Reply:
Its possible to have a (function) method instead of implicit method, but I prefer to keep syntax like:
```
val snapshotter = snapshotStore.asSnapshotter
```

Using any IDE/Metals, one can easily goto definition regardless of its nature. I see this choice as personal preference, but if under the surface there's some technical differences - please share them

Reply:
There are no significant technical differences. I just like to keep the things simple. I.e. to minimize number of implicits/extensions used etc. I love to use simple functions/classes whenever is possible.

I do not think this matters in this case, because API is private, so feel free to keep it the way you want.